### PR TITLE
Allow recipe to change fence timeout.

### PR DESCRIPTION
**Pull request by dj2**
This CL adds an API to the recipe class to set the fence timeout for a
given script.

The default fence timeout is also bumped from 100ms to 1000ms.

### CODE REVIEW

**Comment by dj2** [comment]
@ben-clayton

**Comment by ben-clayton** [comment]
Looks perfect for our needs, thank you!

**Comment by ben-clayton** [comment]
Gentle ping.
This one issue is causing mayhem with our tests. If this is going to take a while, I'll consider adding a timeout statement to all of deqp's amber scripts.

**Comment by dj2** [comment]
Kokoro doesn't seem to be showing the details links anymore so I can't see why the dawn build failed ....

**Comment by sarahM0** [comment]
The failure is because of a new change in dawn buffer UsageBit (not changes in this PR). I'll put up a fix for it shortly.

**Comment by dj2** [comment]
Awesome, thanks Sarah. @dneto0 can you take another look please.

File: src/script.h [hunk 1]
```diff
@@ -58,6 +58,11 @@ class Script : public RecipeImpl {
     return engine_info_.required_instance_extensions;
   }
 
+  /// Sets the fence timeout to |timeout_ms|.
+  void SetFenceTimeout(uint32_t timeout_ms) override {
+    engine_data_.fence_timeout_ms = timeout_ms;
```
[the remark this answers is not in the document]
Comment (dneto0) [reply src/script.h]: ah. thanks

**Pull request merged**